pkg/provenance/sign.go

```diff
@@ -277,12 +277,57 @@ func (s *Signatory) Verify(chartpath, sigpath string) (*Verification, error) {
 	return ver, nil
 }
 
+// VerifyFromData checks the signature and verifies that it is legit for a chart (not stored in the filesystem)
+func (s *Signatory) VerifyFromData(chartRef string, chartData, sigData *bytes.Buffer) (*Verification, error) {
```

Comment:
There is a lot of code duplication between `Verify()` and `VerifyFromData()`.

What you could do is move the first part of the code from `decodeSignature()` into `Verify()` so as to extract the chartData and the copy the first part of `Digest()` in `Verify(), so that you can then have `Verify()` call `VerifyFromData()` passing it the chartData and sigData.

Reply:
I have reduecd the footprint of duplicated code as of now. We would be able to solve the problem with sigData, and chartData but there is an issue of reading this stuff as bytes.Buffer for which we would have to ass some more code.